bpdm-pool/src/main/kotlin/org/eclipse/tractusx/bpdm/pool/repository/LegalEntityRepository.kt

```diff
@@ -37,6 +38,9 @@ interface LegalEntityRepository : PagingAndSortingRepository<LegalEntity, Long>,
 
     fun findByUpdatedAtAfter(updatedAt: Instant, pageable: Pageable): Page<LegalEntity>
 
+    @Query("SELECT p FROM LegalEntity p WHERE LOWER(p.legalName.value) LIKE LOWER(CONCAT('%', :value, '%')) ORDER BY LENGTH(p.legalName.value)")
```

Comment:
In order to increase performance please concatenate the search string before passing it to the repository method.

Reply:
Done